### DISCUSSION TITLE
[useRangeCalendar] Should interacting outside RangeCalendar cancel ongoing selection instead of fulfill it?

### DISCUSSION THREAD

**Discussion by QzCurious** [adobe/react-spectrum#5702] [General]
Just like clicking outside a dialog would effectively close it. So should RangeCalendar, for UX of my.

If it works as design, I would like to know why. Thanks.

https://github.com/adobe/react-spectrum/assets/38932402/9aa5b3d8-ddb0-491e-929f-e2f80ebfc4e0

_The running code is an example from [docs](https://react-spectrum.adobe.com/react-aria/useRangeCalendar.html#styled-examples)_

**Comment by reidbarber** [comment]
The proposed behavior seems reasonable to me. Feel free to convert this into an issue so it can be discussed/prioritized there.

**Comment by QzCurious** [comment]
Seem like I failed to convert it to an issue. FYI, I opened an issue #5709.

**Comment by QzCurious** [comment]
Hi, @reidbarber. I just sorted out the right behavior that I thought. And I made a PR #5721 for it. I tried hard to elaborate all possible use case (the existing ones, ofc) to make it more sensible. Kindly asking you take a look. And I'm happy to put more effort on it if needed. Thanks.
